### PR TITLE
Stable backport fixes on the release workflow for v28.x

### PR DESCRIPTION
**Pull request by likebreath**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -28,13 +28,13 @@ jobs:
         with:
             toolchain: "1.62"
             command: build
-            args: --all --release --target=x86_64-unknown-linux-gnu
+            args: --all --release --no-default-features --features "kvm,mshv" --target=x86_64-unknown-linux-gnu
```
Comment (rbradford) [inline .github/workflows/release.yaml]: Don't add this commit. It might break binary compatibility for migrations.

Comment (likebreath) [reply .github/workflows/release.yaml]: Good point. PR Updated.

**Comment by likebreath** [comment]
Looks like we are hitting some sort of limit from Github for downloading released artifacts. This PR and #5117 failed several times for this reason. Such as:

```
[2023-01-19T18:22:27.103Z] + wget --quiet https://github.com/cloud-hypervisor/cloud-hypervisor/releases/download/v26.0/cloud-hypervisor-static-aarch64 -O cloud-hypervisor-static-aarch64
[2023-01-19T18:22:27.517Z] 
[2023-01-19T18:22:27.517Z] real	0m0.254s
[2023-01-19T18:22:27.517Z] user	0m0.008s
[2023-01-19T18:22:27.517Z] sys	0m0.003s
[2023-01-19T18:22:27.517Z] + exit 1
[2023-01-19T18:22:27.517Z] + RES=1
[2023-01-19T18:22:27.517Z] + '[' 1 -ne 0 ']'
[2023-01-19T18:22:27.517Z] + exit 1
```

and

```
[2023-01-19T19:02:45.808Z] + wget --quiet https://github.com/cloud-hypervisor/rust-hypervisor-firmware/releases/download/0.4.2/hypervisor-fw
[2023-01-19T19:02:45.808Z] ~/workloads /cloud-hypervisor
[2023-01-19T19:02:52.321Z] 
[2023-01-19T19:02:52.321Z] real	0m5.455s
[2023-01-19T19:02:52.321Z] user	0m0.007s
[2023-01-19T19:02:52.321Z] sys	0m0.000s
[2023-01-19T19:02:52.321Z] + exit 1
```

**Comment by rbradford** [comment]
There is an ongoing GitHub outtage right now: https://www.githubstatus.com/incidents/n18j5g7wpg0s but that might be unrelated.

**Comment by likebreath** [comment]
@rbradford PTAL. CI now are happy.

**Comment by rbradford** [comment]
@likebreath Did you create a test tag off this to see if it worked fine?

**Comment by likebreath** [comment]
> Did you create a test tag off this to see if it worked fine?

@rbradford Good point. Running it on my fork now.

**Comment by likebreath** [comment]
@rbradford The release action completed successfully [1]. Want to take a final look before we land it and cut the release?

[1] https://github.com/likebreath/cloud-hypervisor/actions/runs/3988847785/jobs/6840540304

**Review by rbradford — APPROVED** [review]

**Pull request merged**